packages/docusaurus/src/server/plugins/init.ts

```diff
@@ -49,9 +75,23 @@ export default function initPlugins({
       const pluginModule: any = importFresh(
         pluginRequire.resolve(pluginModuleImport),
       );
-      return (pluginModule.default || pluginModule)(context, pluginOptions);
+
+      const plugin = pluginModule.default || pluginModule;
+      if (plugin.validateOptions) {
+        const options = plugin.validateOptions({
+          validate,
+          options: pluginOptions,
+        });
+        pluginOptions = options;
+      }
+      if (plugin.validateThemeConfig) {
+        plugin.validateThemeConfig({
+          validate: validateAndStrip,
+          themeConfig: context.siteConfig.themeConfig,
+        });
+      }
+      return plugin(context, pluginOptions);
```

Comment:
I think we should support plain ES export syntax here:

```
export function validateThemeConfig() { }
```

It does not look like it would work.

We'd need something like:

```
const validateOptions = pluginModule.default?. validateOptions ?? pluginModule.validateOptions;
if ( validateOptions ) {
  ...
}
```

For both plugins and themes.

We could even migrate the existing plugin to ES as well. 

**Note**: we are recommending node >= 10 (not supporting ES imports), so only the TS/transpiled plugins can be migrated.

We should rather document both syntaxes. ES is more modern, but requires newest node versions of 12/14, while Common-js would work as well on older node versions.

Reply:
For documentation, I will add lifecycle methods to docs. And for import, I think we can support both.